File: java-playground/src/main/java/org/gonza/javaplayground/domain/Car.java
package org.gonza.javaplayground.domain;

public class Car {
    private String name;
    private Integer location;

    public static final int THRESHOLD_FOR_MOVING_FORWARD = 4;

    public Car(String name) {
        this.name = name;
        this.location = 1;
    }

    public Integer moveForward() {
        this.location = this.location + 1;
        return this.location;
    }

    public static Boolean isPossibleMovingForward(int n) {
        boolean isPossible = n >= THRESHOLD_FOR_MOVING_FORWARD;
        return isPossible;
    }

    public String getName() {
        return name;
    }

    public Integer getLocation() {
        return location;
    }
}

File: kotlin-playground/src/main/kotlin/org/gonza/kotlinplayground/Car.kt
package org.gonza.kotlinplayground

class Car(
    val name: String,
){
    init {
        require(name.isNotBlank()) { "이름은 비어있을 수 없다" }
    }

    private var _location: Int = 0
    val location: Int
        get() = _location
}
